### PR TITLE
WIP: Update recommended JDK version to 21

### PR DESCRIPTION
**Pull request by agoerler**

### CODE REVIEW

File: java/development/index.md [hunk 1]
```diff
@@ -249,7 +249,7 @@ The CAP Java SDK uses various dependencies that are also used by the application
 
 | Dependency | Minimum Version | Recommended Version |
 | --- | --- | --- |
-| Java | 17 | 17 |
+| Java | 17 | 21 |
```
Comment (mofterdinger) [inline java/development/index.md:252]: Is Java 21 available on BAS ? If we recommend it it should be available on BAS.

**Review by mofterdinger — APPROVED** [review]

File: java/development/index.md (hunk 1 above)
Comment (vmikhailenko) [reply java/development/index.md:252]: Snap! Does not look like it. :(

Comment (vmikhailenko) [reply java/development/index.md]: Ha, `asdf` lists Java 21 as a candidate on trial.

Comment (vmikhailenko) [reply java/development/index.md]: It was presented today. You can switch the Java version on the fly with it. 

```
user: user $ java --version
openjdk 17.0.7 2023-04-18 LTS
OpenJDK Runtime Environment SapMachine (build 17.0.7+7-LTS)
OpenJDK 64-Bit Server VM SapMachine (build 17.0.7+7-LTS, mixed mode, sharing)
user: user $ asdf install java sapmachine-21
############################################################################################################################################################# 100.0%
sapmachine-jdk-21_linux-x64_bin.tar.gz
sapmachine-jdk-21_linux-x64_bin.tar.gz: OK
user: user $ asdf global java sapmachine-21
user: user $ java --version
openjdk 21 2023-09-19 LTS
OpenJDK Runtime Environment SapMachine (build 21+35-LTS)
OpenJDK 64-Bit Server VM SapMachine (build 21+35-LTS, mixed mode, sharing)
user: user $ 
```

Comment (vmikhailenko) [reply java/development/index.md]: But it seems that the configuration for VSCode there is broken.

**Review by renejeglinsky — APPROVED** [review]

**Pull request merged**